cpp/SnowflakeGCSClient.cpp

```diff
@@ -134,12 +146,21 @@ RemoteStorageRequestOutcome SnowflakeGCSClient::GetRemoteFileMetadata(
     url = fileMetadata->presignedUrl;
   }
 
-  if (!m_statement->http_get(url,
-                             reqHeaders,
-                             NULL,
-                             headerString,
-                             true))
+  try
+  {
+    if (!m_statement->http_get(url,
+                               reqHeaders,
+                               NULL,
+                               headerString,
+                               true))
+    {
+      return RemoteStorageRequestOutcome::FAILED;
```

Comment:
FAILED can happend when file exist or some IOException happen. How to differentiate it? right now as coded any exception treat as file exist and just SKIPPED.

Reply:
SKIPPED would be the case when SUCCESS on getting metadata. If there is anything else wrong the uploading would fail with the same reason and error could be thrown there.